plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java

```diff
@@ -3305,6 +3306,23 @@ private static void assertProjectionResult(Optional<ProjectionApplicationResult<
         assertEquals(actualAssignments.size(), expectedAssignments.size());
     }
 
+    @Test
+    public void testApplyRedirection()
+            throws Exception
+    {
+        SchemaTableName tableName = temporaryTable("apply_redirection_tester");
+        doCreateEmptyTable(tableName, ORC, CREATE_TABLE_COLUMNS);
+        try (Transaction transaction = newTransaction()) {
+            ConnectorSession session = newSession();
+            ConnectorMetadata metadata = transaction.getMetadata();
+            ConnectorTableHandle tableHandle = getTableHandle(metadata, tableName);
+            assertThat(metadata.applyTableScanRedirect(session, tableHandle)).isEmpty();
```

Comment:
nit: instead you could pass mock redirection provided that would redirect for `apply_redirection_tester`